File: leet_PREFIX_1921_kill_monsters.py
class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        time = [(d + s - 1) // s for d, s in zip(dist, speed)]
        time.sort()
        res = 0
        for t in time:
            if res == t:
                break
            res += 1
        return res

File: leet_PREFIX_1921_kill_monsters.cc
class Solution {
public:
    int eliminateMaximum(vector<int>& dist, vector<int>& speed)
    {
        vector<int> T;
        int len = (int) dist.size();
        int i = -1;
        while (++i < len)
            T.push_back((dist[i] + speed[i] - 1) / speed[i]);
        sort(T.begin(), T.end());
        int res = 0;
        for (int & t : T)
        {
            if (res == t)
                break ;
            res++;
        }
        return res;
    }
};
